internal/ingest/upload.go

```diff
@@ -117,3 +137,61 @@ func (w *goaWrapper) initSIP(
 
 	return nil
 }
+
+func (w *goaWrapper) findOrCreateUser(
+	ctx context.Context,
+	perSvc persistence.Service,
+) (*datatypes.User, error) {
+	claims := auth.UserClaimsFromContext(ctx)
+	if claims == nil {
+		return nil, nil
+	}
+	if claims.ISS == "" {
+		return nil, fmt.Errorf("invalid user claims: missing ISS")
+	}
+	if claims.Sub == "" {
+		return nil, fmt.Errorf("invalid user claims: missing Sub")
+	}
+
+	user, err := perSvc.ReadJWTUser(ctx, claims.ISS, claims.Sub)
+	if err != nil {
+		if errors.Is(err, persistence.ErrNotFound) {
+			// User does not exist, create a new one.
+			user, err := w.createUser(ctx, perSvc, claims)
```

Comment:
Same here, `w.createUser` will have access to `w.perSvc`, so no need to pass it.